cli/attach.go

```diff
@@ -0,0 +1,110 @@
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/alibaba/pouch/pkg/ioutils"
+	"github.com/alibaba/pouch/pkg/term"
+	"github.com/sirupsen/logrus"
+)
+
+// AttachDescription is used to describe attach command in detail and auto generate command doc.
+var AttachDescription = "Attach local standard input, output, and error streams to a running container"
+
+// AttachCommand is used to implement 'attach' command.
+type AttachCommand struct {
+	baseCommand
+
+	// flags for attach command
+	noStdin    bool
+	detachKeys string
+}
+
+// Init initialize "attach" command.
+func (ac *AttachCommand) Init(c *Cli) {
+	ac.cli = c
+	ac.cmd = &cobra.Command{
+		Use:   "attach [OPTIONS] CONTAINER",
+		Short: "Attach local standard input, output, and error streams to a running container",
+		Long:  AttachDescription,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return ac.runAttach(args)
+		},
+		Example: ac.example(),
+	}
+	ac.addFlags()
+}
+
+// addFlags adds flags for specific command.
+func (ac *AttachCommand) addFlags() {
+	flagSet := ac.cmd.Flags()
+	flagSet.BoolVar(&ac.noStdin, "no-stdin", false, "Do not attach STDIN")
+	flagSet.StringVar(&ac.detachKeys, "detach-keys", "", "Override the key sequence for detaching a container")
+}
+
+// runAttach is used to attach a container.
+func (ac *AttachCommand) runAttach(args []string) error {
+	name := args[0]
+
+	ctx := context.Background()
+	apiClient := ac.cli.Client()
+
+	in, out, err := setRawMode(!ac.noStdin, false)
+	if err != nil {
+		return fmt.Errorf("failed to set raw mode")
+	}
+	defer func() {
+		if err := restoreMode(in, out); err != nil {
+			fmt.Fprintf(os.Stderr, "failed to restore term mode")
+		}
+	}()
+
+	conn, br, err := apiClient.ContainerAttach(ctx, name, !ac.noStdin)
+	if err != nil {
+		return fmt.Errorf("failed to attach container: %v", err)
+	}
+	defer conn.Close()
+
+	var inReader io.Reader = os.Stdin
+	escapeKeys := term.DefaultEscapeKeys
+	// Wrap the input to detect detach escape sequence.
+	// Use default escape keys if an invalid sequence is given.
+	if ac.detachKeys != "" {
+		customEscapeKeys, err := term.ToBytes(ac.detachKeys)
+		if err != nil {
+			logrus.Warnf("invalid detach escape keys, using default: %s", err)
+		} else {
+			escapeKeys = customEscapeKeys
+		}
+	}
+	inReader = ioutils.NewReadCloserWrapper(term.NewEscapeProxy(os.Stdin, escapeKeys), os.Stdin.Close)
+
+	wait := make(chan struct{})
+	go func() {
+		io.Copy(os.Stdout, br)
+		wait <- struct{}{}
+	}()
+	go func() {
+		_, err := io.Copy(conn, inReader)
+		if _, ok := err.(term.EscapeError); ok {
+			wait <- struct{}{}
+		}
+	}()
+
+	<-wait
+	return nil
+}
+
+// example shows examples in attach command, and is used in auto-generated cli docs.
+func (ac *AttachCommand) example() string {
+	return `$ pouch ps
+Name     ID       Status    Image                              Runtime
+foo      71b9c1   Running   docker.io/library/busybox:latest   runc
+$ pouch attach foo`
```

Comment:
Will this command display the result?

Reply:
Sometimes the attach command don't get the result obviously, it depends on the how you use it.